homeassistant/components/ads/__init__.py

```diff
@@ -273,14 +286,20 @@ def __init__(self, ads_hub, name, ads_var):
         self._event = None
 
     async def async_initialize_device(
-        self, ads_var, plctype, state_key=STATE_KEY_STATE, factor=None
+        self, ads_var, plctype, digit=None, state_key=STATE_KEY_STATE, factor=None
     ):
         """Register device notification."""
 
         def update(name, value):
             """Handle device notifications."""
             _LOGGER.debug("Variable %s changed its value to %d", name, value)
 
+            """Format the float, with the Paramter digit"""
+            if plctype == pyads.PLCTYPE_REAL:
+                value = float("{:0.{prec}f}".format(value, prec = digit))
```

Comment:
Please if possible use f-strings instead of str.format.

Reply:
I don't have much experience with python yet, but i think this is not possible, because the value is a float an not a string. If I am wrong, please give me an input, how i can do this.

Reply:
Here's the stackoverflow answer that shows how to do it:
https://stackoverflow.com/a/50340297

Reply:
Now I use f-string.